deepparse/comparer/address_comparer.py

```diff
@@ -0,0 +1,170 @@
+from deepparse.parser import AddressParser
+from typing import List, Union
+
+
+class AdressComparer:
+    """
+        Compares addresses with each others and retrieve the the differences between them.
+    """
+    
+    def __init__(self, parser: AddressParser) -> None:
+        """
+        Address parser used to parse the addresses
+        """
+
+        self.parser = parser
+
+    def __str__(self) -> str:
+        return f"Compare addresses with {self.parser.model_type.capitalize()}AddressParser"
+
+    __repr__ = __str__  # to call __str__ when list of address
+    
+    def compare_with_deepparse(self, addresses_to_compare: Union[List[List[tuple]], List[tuple]]) -> dict:
```

Comment:
```suggestion
    def compare_addresses_tags(self, addresses_to_compare: Union[List[List[tuple]], List[tuple]]) -> dict:
```

Reply:
What we really compare is either or not the tags are the same between an already parsed address and our parsing.